src/Http/Ownership/RequestOwnershipRequestHandler.php

```diff
@@ -51,6 +57,22 @@ public function handle(ServerRequestInterface $request): ResponseInterface
         /** @var RequestOwnership $requestOwnership */
         $requestOwnership = $requestBodyParser->parse($request)->getParsedBody();
 
+        // Make sure there is no open request for this item and owner
+        try {
+            $ownershipItem = $this->ownershipSearchRepository->getByItemIdAndOwnerId(
+                $requestOwnership->getItemId()->toString(),
+                $requestOwnership->getOwnerId()->toString()
+            );
+        } catch (OwnershipItemNotFound $e) {
+            $ownershipItem = null;
```

Comment:
Why the extra step of putting it to null and then throwing an error? getByItemIdAndOwnerId() can never return a null, so this is the only case the check below could trigger.

Reply:
Indeed, this flow can be improved, see these changes https://github.com/cultuurnet/udb3-backend/pull/1652/commits/5f1c3f1b67d093ccb7f391f126cc114194466ed7